src/client/components/RetroCertsRoute/index.js

```diff
@@ -1,19 +1,60 @@
 import PropTypes from "prop-types";
-import { Route } from "react-router-dom";
+import { Route, Redirect } from "react-router-dom";
 import React from "react";
 import { userDataPropType, setUserDataPropType } from "../../commonPropTypes";
+import AUTH_STRINGS from "../../../data/authStrings";
 import PageNotFound from "../../pages/PageNotFound";
 
+function userIsAuthenticated(userData) {
+  return !!userData.weeksToCertify;
+}
+
 function RetroCertsRoute(props) {
-  const { pageComponent: Component, isProduction, pageProps, ...routeProps } = props;
+  const {
+    pageComponent: Component,
+    isProduction,
+    pageProps,
+    requiresAuthentication,
+    ...routeProps } = props;
 
-  // TODO: Move authentication logic into this component.
+  let routeChild = <div>Loading...</div>;
+  if (isProduction) {
+    routeChild = <PageNotFound />;
+  } else if (!requiresAuthentication || userIsAuthenticated(pageProps.userData)) {
+    routeChild = <Component {...pageProps} />;
+  } else {
+    // eslint-disable-next-line no-console
+    console.log('user not authenticated');
```

Comment:
in case it's not intentional you could use console.error to avoid having to escape eslint?

Reply:
Oops, not intentional! Sorry I missed this comment, I'll remove in my next change.